dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/taskgroup/TaskGroupCoordinator.java

```diff
@@ -198,10 +222,30 @@ private void dealWithForceStartTaskGroupQueue() {
         // Find the force start task group queue(Which is inQueue and forceStart is YES)
         // Notify the related waiting task instance
         // Set the taskGroupQueue status to RELEASE and remove it from queue
-        List<TaskGroupQueue> taskGroupQueues = taskGroupQueueDao.queryAllInQueueTaskGroupQueue()
-                .stream()
-                .filter(taskGroupQueue -> Flag.YES.getCode() == taskGroupQueue.getForceStart())
-                .collect(Collectors.toList());
+        // We use limit here to avoid OOM, and we will retry to notify force start queue at next time
+        int minTaskGroupQueueId = -1;
+        int limit = DEFAULT_LIMIT;
```

Comment:
Can this `limit` be modified by user or just a fixed value?

Reply:
A fixed value is ok, this config does not need to be exposed to the user.